src/scales/scale.time.js

```diff
@@ -226,6 +290,22 @@ function determineMajorUnit(unit) {
 	}
 }
 
+function addTick(scale, ticks, time) {
+	if (scale.options.ticks.source !== 'data') {
+		ticks.add(time);
+		return;
+	}
+
+	const timestamps = getDataTimestamps(scale);
```

Comment:
Another odd thing to have in inner loop. We are still potentially generating a lot of ticks, depending on the unit.

Reply:
updated